src/runtime/GlobalObjectBuiltinString.cpp

```diff
@@ -553,6 +553,82 @@ static Value builtinStringReplace(ExecutionState& state, Value thisValue, size_t
     }
 }
 
+static Value builtinStringReplaceAll(ExecutionState& state, Value thisValue, size_t argc, Value* argv, Optional<Object*> newTarget)
+{
+    if (thisValue.isUndefinedOrNull()) {
+        ErrorObject::throwBuiltinError(state, ErrorObject::TypeError, state.context()->staticStrings().object.string(), true, state.context()->staticStrings().replace.string(), ErrorObject::Messages::GlobalObject_ThisUndefinedOrNull);
+    }
+    Value searchValue = argv[0];
+    Value replaceValue = argv[1];
+    // If searchValue is neither undefined nor null, then
+    if (!searchValue.isUndefinedOrNull()) {
+        bool isSearchValueRegExp = searchValue.isObject() && searchValue.asObject()->isRegExp(state);
+        // If isRegExp is true, then
+        if (isSearchValueRegExp) {
+            String* flags = searchValue.asObject()->get(state, ObjectPropertyName(state, state.context()->staticStrings().flags)).value(state, searchValue).toString(state);
+            if (flags->find("g") == SIZE_MAX) {
```

Comment:
>  Let flags be ? Get(searchValue, "flags").
    Perform ? RequireObjectCoercible(flags).
    If ? ToString(flags) does not contain "g", throw a TypeError exception.

According to the standard, we needs to check the type of `flags` value before calling `ToString` method.

And one more thing, `contains` would be more intuitive instead of `find` method at here.